implementations/rust/ockam/ockam_api/src/auth.rs

```diff
@@ -0,0 +1,317 @@
+pub mod store;
+pub mod types;
+
+use crate::{Error, Method, Request, RequestBuilder, Response, Status};
+use core::convert::Infallible;
+use core::fmt;
+use minicbor::encode::Write;
+use minicbor::{Decoder, Encode};
+use ockam_core::compat::error::Error as StdError;
+use ockam_core::compat::io;
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{self, Address, Route, Routed, Worker};
+use ockam_node::Context;
+use store::Storage;
+use tracing::{trace, warn};
+use types::{Attribute, Attributes};
+
+/// Auth API server.
+#[derive(Debug)]
+pub struct Server<S> {
+    store: S,
+}
+
+#[ockam_core::worker]
+impl<S: Storage + Send + Sync + 'static> Worker for Server<S> {
```

Comment:
Pros vs Cons adding `Send + Sync + 'static` as a requirement for the Storage trait itself? Not sure we ever use it without this requirement

Reply:
Still more general to constrain it on call site I think.